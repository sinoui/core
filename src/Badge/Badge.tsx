import React from 'react';
import styled, { css } from 'styled-components';
import classNames from 'classnames';
import colorCss from '@sinoui/core/utils/colorCss';

export interface BadgeProps {
  /**
   * 子元素
   */
  children?: React.ReactNode;
  /**
   * 添加自定义类名
   */
  className?: string;
  /**
   * 指定颜色
   */
  color?: string;
  /**
   * 设置为圆点形式
   */
  dot?: boolean;
  /**
   * 	指定数字
   */
  count?: number;
  /**
   * 	指定封顶数值
   */
  overflowCount?: number;
  /**
   * 	设置数字为0时显示
   */
  showZero?: boolean;
  /**
   * 	设置自定义badge的内容
   */
  badgeContent?: React.ReactNode;
  /**
   * 	设置鼠标悬浮时显示的文字
   */
  title?: string;
  /**
   * 	设置徽标的显示位置
   */
  anchorOrigin?: { horizontal: 'left' | 'right'; vertical: 'bottom' | 'top' };
}

const BadgeWrapper = styled.span`
  display: inline-flex;
  position: relative;
  flex-shrink: 0;
  vertical-align: middle;
`;

const dotCss = css`
  width: 8px;
  height: 8px;
`;

const overflowCountCss = css`
  border-radius: 10px;
  padding: 0px 6px;
`;

const anchorOriginCss = css<BadgeProps>`
  ${(props) => props.anchorOrigin && `${props.anchorOrigin.vertical}: 0`};
  ${(props) => props.anchorOrigin && `${props.anchorOrigin.horizontal}: 0`};
  ${(props) =>
    props.anchorOrigin &&
    `transform: scale(1) translate(${
      props.anchorOrigin.horizontal === 'right' ? '50%' : '-50%'
    }, ${props.anchorOrigin.vertical === 'top' ? '-50%' : '50%'})`};
  ${(props) =>
    props.anchorOrigin &&
    `transform-origin: ${
      props.anchorOrigin.horizontal === 'right' ? '100%' : '0%'
    } ${props.anchorOrigin.vertical === 'top' ? '0%' : '100%'}`}
`;

const BadgeCircle = styled.span<BadgeProps>`
  position: absolute;
  ${anchorOriginCss};
  min-width: ${(props) => !props.dot && '20px'};
  height: 20px;
  border-radius: 50%;
  ${colorCss('background-color')};
  color: #fff;
  font-size: ${(props) => props.theme.typography.caption};
  display: flex;
  justify-content: center;
  text-align: center;
  ${(props) => props.dot && dotCss};
  ${(props) =>
    props.count &&
    props.overflowCount &&
    props.count > props.overflowCount &&
    overflowCountCss};
`;

const Wrapper = styled.div`
  position: absolute;
  ${anchorOriginCss};
`;

/**
 * 徽章 组件
 *
 * @param {BadgeProps} props
 * @returns
 */
const Badge = React.forwardRef(
  (props: BadgeProps, ref: React.Ref<HTMLInputElement>) => {
    const {
      children,
      className,
      dot,
      count,
      color = 'error',
      showZero,
      title,
      badgeContent,
      overflowCount = 99,
      anchorOrigin = { vertical: 'top', horizontal: 'right' },
      ...rest
    } = props;

    const num =
      overflowCount && count && count > overflowCount
        ? `${overflowCount}+`
        : count;
    return (
      <BadgeWrapper
        ref={ref}
        className={classNames('sinoui-badge', className, {
          'sinoui-badge--dot': dot,
        })}
      >
        {badgeContent ? (
          <Wrapper
            className="sinoui-badge__content"
            anchorOrigin={anchorOrigin}
          >
            {badgeContent}
          </Wrapper>
        ) : (
          ((count && count > 0) || showZero || dot) && (
            <BadgeCircle
              color={color}
              dot={dot}
              title={title}
              overflowCount={overflowCount}
              anchorOrigin={anchorOrigin}
              count={count}
              {...rest}
            >
              {!dot ? num : null}
            </BadgeCircle>
          )
        )}
        {children}
      </BadgeWrapper>
    );
  },
);

export default Badge;
