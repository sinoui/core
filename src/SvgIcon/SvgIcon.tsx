import React from 'react';
import styled from 'styled-components';
import classNames from 'classnames';
import colorCss from '@sinoui/core/utils/colorCss';
import { removeUndefinedProperties } from '../utils/objects';

export interface SvgIconProps {
  /**
   * 指定图标内容
   */
  children?: React.ReactNode;
  /**
   * 自定义类名
   */
  className?: string;
  /**
   * 指定自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 设置 svg 元素的 viewBox，默认为0 0 24 24
   */
  viewBox?: string;
  /**
   * 设置图标大小。数字或者字符串。如果是数字，则单位是像素
   */
  size?: number | string;
  /**
   * 设置颜色
   */
  color?: string;
  /**
   * 指定图标的标题
   */
  title?: string;
  /**
   * 不设置图标大小。一般用于`<SvgIcon as={CustomIcon} disabledViewBox />`，这种情况下，CustomIcon已有了`viewBox`。
   */
  disabledViewBox?: boolean;
}

/**
 * SvgIcon组件
 */
const SvgIcon = styled.svg
  .withConfig({
    shouldForwardProp: (prop) => !['disabledViewBox'].includes(prop),
  })
  .attrs(
    ({
      children,
      className,
      disabledViewBox,
      viewBox = !disabledViewBox ? '0 0 24 24' : undefined,
      size = 24,
      title,
    }: SvgIconProps) =>
      removeUndefinedProperties({
        children: (
          <>
            {!!title && <title>{title}</title>}
            {children}
          </>
        ),
        className: classNames('sinoui-svg-icon', className),
        viewBox,
        size,
        'aria-hidden': title ? undefined : 'true',
        role: title ? 'img' : 'presentation',
        focusable: 'false',
      }),
  )<SvgIconProps>`
  font-size: ${(props) =>
    typeof props.size === 'string' ? props.size : `${props.size}px`};
  width: 1em;
  height: 1em;
  fill: currentColor;
  flex-shrink: 0;
  transition: ${(props) =>
    props.theme.transitions.create('color', {
      duration: props.theme.transitions.duration.shorter,
    })};
  ${colorCss('color')};
  user-select: none;
  display: inline-block;
`;

export default SvgIcon;
