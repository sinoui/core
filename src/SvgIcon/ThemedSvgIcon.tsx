import styled from 'styled-components';
import { getColorFromTheme } from '../utils/colors';
import { ColorProp } from '../types';
import SvgIcon from './SvgIcon';

export interface Props {
  color?: ColorProp;
}

const ThemedSvgIcon = styled(SvgIcon) <Props>`
  ${(props) =>
    props.color !== 'none' && `color: ${getColorFromTheme(props, 'primary')}`};
`;

ThemedSvgIcon.sinouiName = 'SvgIcon';
ThemedSvgIcon.displayName = 'SvgIcon';

export default ThemedSvgIcon;
