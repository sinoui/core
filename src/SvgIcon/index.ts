import SvgIcon, { SvgIconProps } from './SvgIcon';

export type { SvgIconProps };
export default SvgIcon;
