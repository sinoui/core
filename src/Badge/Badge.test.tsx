import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import SvgIcon from '@sinoui/core/SvgIcon';
import { MdEmail } from 'react-icons/md';
import Badge from '@sinoui/core/Badge';

/**
 * Badge组件 单元测试
 */

describe('Badge组件 单元测试', () => {
  afterEach(cleanup);

  it('设置为圆点形式', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Badge count={1} dot>
          <SvgIcon as={MdEmail} />
        </Badge>
      </ThemeProvider>,
    );

    const text = container.querySelector('.sinoui-badge');
    expect(text).toHaveClass('sinoui-badge--dot');
  });

  it('指定数字', () => {
    const { getByText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Badge count={1}>
          <SvgIcon as={MdEmail} />
        </Badge>
      </ThemeProvider>,
    );

    expect(getByText('1')).toBeInTheDOM();
  });

  it('指定封顶数值', () => {
    const { getByText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Badge count={100} overflowCount={99}>
          <SvgIcon as={MdEmail} />
        </Badge>
      </ThemeProvider>,
    );

    expect(getByText('99+')).toBeInTheDOM();
  });

  it('设置徽标的显示位置', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Badge count={8} anchorOrigin={{ vertical: 'top', horizontal: 'left' }}>
          <SvgIcon as={MdEmail} />
        </Badge>
      </ThemeProvider>,
    );

    const text = container.querySelector('.sinoui-badge');
    expect(text?.firstChild).toHaveStyle('top:0;left:0');
  });
});

describe('Badge组件 快照测试', () => {
  it('基本使用', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge>
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('指定数字', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge count={8}>
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('设置数字为0时显示', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge count={0} showZero>
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('设置为圆点形式', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge count={1} dot>
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('设置鼠标悬浮时显示的文字', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge count={1} title="文字">
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('设置自定义badge的内容', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge count={1} badgeContent={4}>
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('指定封顶数值', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge count={99} overflowCount={99}>
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge count={100} overflowCount={99}>
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('设置徽标的显示位置', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge
            count={8}
            anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
          >
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge
            count={8}
            anchorOrigin={{ vertical: 'top', horizontal: 'left' }}
          >
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge
            count={8}
            anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
          >
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge
            count={8}
            anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
          >
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('指定颜色', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Badge count={99} color="primary">
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge count={99} color="secondary">
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge count={99} color="warning">
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge count={99} color="success">
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge count={99} color="error">
            <SvgIcon as={MdEmail} />
          </Badge>
          <Badge count={99} color="info">
            <SvgIcon as={MdEmail} />
          </Badge>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
