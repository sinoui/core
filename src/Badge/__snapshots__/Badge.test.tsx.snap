// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge组件 快照测试 基本使用 1`] = `
<span
  className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
>
  <svg
    aria-hidden="true"
    className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
    fill="currentColor"
    focusable="false"
    height={24}
    role="presentation"
    size={24}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
    />
  </svg>
</span>
`;

exports[`Badge组件 快照测试 指定封顶数值 1`] = `
Array [
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 iUpzoi"
      color="error"
    >
      99
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 fgEXpB"
      color="error"
    >
      99+
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
]
`;

exports[`Badge组件 快照测试 指定数字 1`] = `
<span
  className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
>
  <span
    className="Badge__BadgeCircle-sc-1mox8ry-1 iUpzoi"
    color="error"
  >
    8
  </span>
  <svg
    aria-hidden="true"
    className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
    fill="currentColor"
    focusable="false"
    height={24}
    role="presentation"
    size={24}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
    />
  </svg>
</span>
`;

exports[`Badge组件 快照测试 指定颜色 1`] = `
Array [
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 kQonTs"
      color="primary"
    >
      99
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 ehiKHD"
      color="secondary"
    >
      99
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 fbsKxv"
      color="warning"
    >
      99
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 kRCPyf"
      color="success"
    >
      99
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 iUpzoi"
      color="error"
    >
      99
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 eIXhYb"
      color="info"
    >
      99
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
]
`;

exports[`Badge组件 快照测试 设置为圆点形式 1`] = `
<span
  className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge sinoui-badge--dot"
>
  <span
    className="Badge__BadgeCircle-sc-1mox8ry-1 irelgB"
    color="error"
  />
  <svg
    aria-hidden="true"
    className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
    fill="currentColor"
    focusable="false"
    height={24}
    role="presentation"
    size={24}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
    />
  </svg>
</span>
`;

exports[`Badge组件 快照测试 设置徽标的显示位置 1`] = `
Array [
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 iUpzoi"
      color="error"
    >
      8
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 gvQXZ"
      color="error"
    >
      8
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 fPEXWi"
      color="error"
    >
      8
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
  <span
    className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
  >
    <span
      className="Badge__BadgeCircle-sc-1mox8ry-1 imBWBZ"
      color="error"
    >
      8
    </span>
    <svg
      aria-hidden="true"
      className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
      fill="currentColor"
      focusable="false"
      height={24}
      role="presentation"
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>,
]
`;

exports[`Badge组件 快照测试 设置数字为0时显示 1`] = `
<span
  className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
>
  <span
    className="Badge__BadgeCircle-sc-1mox8ry-1 kYsYRK"
    color="error"
  >
    0
  </span>
  <svg
    aria-hidden="true"
    className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
    fill="currentColor"
    focusable="false"
    height={24}
    role="presentation"
    size={24}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
    />
  </svg>
</span>
`;

exports[`Badge组件 快照测试 设置自定义badge的内容 1`] = `
<span
  className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
>
  <div
    className="Badge__Wrapper-sc-1mox8ry-2 cAjkyo sinoui-badge__content"
  >
    4
  </div>
  <svg
    aria-hidden="true"
    className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
    fill="currentColor"
    focusable="false"
    height={24}
    role="presentation"
    size={24}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
    />
  </svg>
</span>
`;

exports[`Badge组件 快照测试 设置鼠标悬浮时显示的文字 1`] = `
<span
  className="Badge__BadgeWrapper-sc-1mox8ry-0 ljnePt sinoui-badge"
>
  <span
    className="Badge__BadgeCircle-sc-1mox8ry-1 iUpzoi"
    color="error"
    title="文字"
  >
    1
  </span>
  <svg
    aria-hidden="true"
    className="SvgIcon-sc-1015m7s-0 kTyazq sinoui-svg-icon"
    fill="currentColor"
    focusable="false"
    height={24}
    role="presentation"
    size={24}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
    />
  </svg>
</span>
`;
